src/App.js

```diff
@@ -1,16 +1,43 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import chatMessages from './data/messages.json';
+import ChatEntry from './components/ChatEntry';
+import ChatLog from './components/ChatLog';
+
 
 const App = () => {
+
+  const [chatData, setChatData] = useState(chatMessages);
+  let [likeCount, setLikeCount] = useState(0);
+  // let likeCount = 0;
+
+  const onLikeClick = (id) => {
+  setChatData(chatData.map(chat => {
+    if (chat.id === id){
+      chat.liked = !chat.liked
+      if (chat.liked === true){
+        setLikeCount(likeCount += 1)
+      };
+      if (chat.liked === false){
+        setLikeCount(likeCount -= 1)
+      };
```

Comment:
Our state `chatData` contains info on whether a message is liked or not, which means we could derive our heart count from the data by looping over `chatData` and counting up how many entries have a `liked` value of `True`. This would let us remove the `likeCount` state that we need to manually keep in sync with the changes to `chatData`. What changes would we need to make for that to work?